app/models/workarea/catalog/product.decorator

```diff
@@ -1,7 +1,25 @@
 module Workarea
   decorate Catalog::Product, with: 'affirm' do
     decorated do
-      field :affirm_available, type: Boolean, default: true
+      field :affirm_available, type: Boolean
```

Comment:
It's unusual to set a field value in a validate method, usually we separate that (CQS). Would something like this work?

```
field :affirm_available, type: Boolean, default: -> { default_affirm_available }

def default_affirm_available
  try(:gift_card?) ? false : true
end

def gift_card_cannot_be_purchased_using_affirm
  if  try(:gift_card?) && affirm_available?
    cannot_be_set_for_gift_cards = I18n.t(
      'workarea.admin.catalog_products.affirm.cannot_be_set_for_gift_cards'
    )

    errors.add(:affirm_available, cannot_be_set_for_gift_cards)
  end
end
```

Reply:
love it, much easier to understand what's going on.